sink.go

```diff
@@ -45,6 +45,7 @@ type Sink interface {
 type sinkRegistry struct {
 	mu        sync.Mutex
 	factories map[string]func(*url.URL) (Sink, error) // keyed by scheme
+	openFile  func(string, int, os.FileMode) (*os.File, error)
```

Comment:
consider a comment adding that this is the same signature as os.OpenFile.

```suggestion
	openFile  func(string, int, os.FileMode) (*os.File, error) // == os.OpenFile
```